src/main/java/org/broadinstitute/hellbender/engine/filters/CountingReadFilter.java

```diff
@@ -0,0 +1,179 @@
+package org.broadinstitute.hellbender.engine.filters;
+
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.function.Predicate;
+import java.util.stream.IntStream;
+
+/**
+ * Wrapper/adapter for ReadFilter that counts the number of reads filtered, and provides a filter count summary.
+ *
+ * For filters with complex predicates that are composed of compound and/or operators, provides counts
+ * by level for each component predicate.
+ *
+ */
+public class CountingReadFilter implements ReadFilter {
+
+    static private final long serialVersionUID = 1L;
+
+    // Underlying ReadFilter we delegate to if we're wrapping a simple ReadFilter
+    protected final ReadFilter delegateFilter;
+
+    // Name for this filter, used to report filter count summaries
+    protected final String name;
+
+    // Number of reads filtered by this filter
+    protected long filteredCount = 0;
+
+    public CountingReadFilter(final String filterName, final ReadFilter readFilter) {
+        Utils.nonNull(readFilter);
+        name = filterName;
+        delegateFilter = readFilter;
+    }
+
+    // Used by the private CountingBinopReadFilter subclass
+    private CountingReadFilter(final String filterName) {
+        name = filterName;
+        delegateFilter = null;
+    }
+
+    // Return the number of reads filtered by this filter
+    public long getFilteredCount() {
+        return filteredCount;
+    }
+
+    public void resetFilteredCount() {
+        filteredCount = 0;
+    }
+
+    public String getName() {return name;}
+
+    // Returns a summary line with filter counts organized by level
+    public String getSummaryLine() {return getSummaryLineForLevel(0);}
+
+    protected String getSummaryLineForLevel(final int indentLevel) {
+        if (0 == filteredCount) {
+            return 0 == indentLevel ? "" : "No reads filtered by: " + getName();
```

Comment:
I'd vote for not suppressing any output for now and just always returning the `"No reads filtered by: "` text when `filteredCount == 0`. We can tweak the output later if it proves annoying or confusing.